Basic server side export API for saved objects (#30326)
elastic/kibana: docs/api/saved-objects/export_by_type.asciidoc

[[saved-objects-api-export-by-type]]
=== Export Objects By Type

experimental[This functionality is *experimental* and may be changed or removed completely in a future release.]

> The export saved objects by type API enables you to retrieve a set of Kibana saved objects into a newline delimited JSON format.

Reviewer: Maybe we should avoid mentioning the format here, since we intend the format to be mostly irrelevant to people. How about:
```suggestion
The export saved objects by type API enables you to retrieve a set of saved objects that can later be imported into Kibana.
```

Note: You cannot access this endpoint via the Console in Kibana.

==== Request

`POST /api/saved_objects/_export_by_type/{type}`

==== Response body

> The response body will have a format of newline delimited JSON. The order of these objects is not guaranteed other than having dependencies defined first.

Reviewer: I thought we had planned on sorting the objects when we request them from Elasticsearch so that then they are sorted with dependencies first they would always be in the same order?

==== Examples

> The following example attempts to export index patterns and visualizations

Reviewer: ```suggestion
The following example exports all index pattern and visualization saved objects.
```
